File: mock-backend/mock/src/main/java/com/group/mock/service/ProfileService.java
package com.group.mock.service;

import com.group.mock.entity.DTO.request.UpdateProfileRequest;
import com.group.mock.entity.DTO.request.UpdateWorkerProfileRequest;
import com.group.mock.entity.DTO.response.MyProfileResponse;
import com.group.mock.entity.DTO.response.PublicWorkerProfileResponse;
import java.util.UUID;

public interface ProfileService {

    MyProfileResponse getMyProfile(String username);

    MyProfileResponse updateMyProfile(String username, UpdateProfileRequest request);

    MyProfileResponse updateMyWorkerProfile(String username, UpdateWorkerProfileRequest request);

    PublicWorkerProfileResponse getPublicWorkerProfile(UUID workerId, String viewerUsername);
}

File: mock-frontend/src/services/profileService.js
import axiosClient from '../api/axiosClient'

export function getMyProfile() {
  return axiosClient.get('/api/v1/profile/me')
}

export function updateMyProfile(payload) {
  return axiosClient.put('/api/v1/profile/me', payload)
}

export function updateMyWorkerProfile(formData) {
  return axiosClient.put('/api/v1/profile/me/worker', formData)
}

export function getPublicWorkerProfile(workerId) {
  return axiosClient.get(`/api/v1/workers/${workerId}`)
}
